### PR TITLE
[Docathon] Fix NO.30-NO.40 API label

### PR DESCRIPTION
**Pull request by zade23**
### PR types

Others

### PR changes

Docs 

### Description
#57363

| 序号 | Origin label name                       | Fixed  label name                             |
| ---- | --------------------------------------- | --------------------------------------------- |
| 30   | `api_base_transpiler_HashName`          | `api_fluid_transpiler_HashName`               |
| 31   | `api_base_transpiler_RoundRobin`        | `api_fluid_transpiler_RoundRobin`             |
| 32   | `api_fluid_ParamAttr`                   | `api_paddle_ParamAttr`                        |
| 33   | `api_initializer_NumpyArrayInitializer` | `api_paddle_nn_initializer_Assign` |
| 34   | `api_logical_xor`                       | `api_paddle_logical_xor`                      |
| 35   | `api_nn_Bilinear`                       | `api_paddle_nn_Bilinear`                       |
| 36   | `api_nn_cn_elu`                         | `api_paddle_nn_ELU`                           |
| 37   | `api_nn_cn_relu`                        | `api_paddle_nn_ReLU`                          |
| 38   | `api_nn_cn_softmax`                     | `api_paddle_nn_Softmax`                       |
| 39   | `api_nn_conv_ConvTranspose2d`           | `api_paddle_nn_Conv2DTranspose`   |
| 40   | `api_nn_conv_ConvTranspose3d`           | `api_paddle_nn_Conv3DTranspose`   |

> 其中序号32出现了4次，不知道是不是漏了一次

 @ooooo-create @sunzhongkai588

### CODE REVIEW

File: python/paddle/distributed/fleet/layers/mpu/mp_layers.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class VocabParallelEmbedding(paddle.nn.Layer):
             default weight parameter property is used. See usage for details in :ref:`api_ParamAttr` . In addition,
             user-defined or pre-trained word vectors can be loaded with the :attr:`param_attr` parameter.
             The local word vector needs to be transformed into numpy format, and the shape of local word
-            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_initializer_NumpyArrayInitializer`
+            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_fluid_initializer_NumpyArrayInitializer`
```
Comment (ooooo-create) [inline python/paddle/distributed/fleet/layers/mpu/mp_layers.py]:
```suggestion
            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_paddle_nn_initializer_Assign`
```

File: python/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -903,7 +903,7 @@ def bilinear(x1, x2, weight, bias=None, name=None):
     """
 
     This layer performs bilinear on two inputs.
-    See :ref:`api_nn_Bilinear` for details and output shape.
+    See :ref:`api_paddle_nn_Biliner` for details and output shape.
```
Comment (ooooo-create) [inline python/paddle/nn/functional/common.py]:
```suggestion
    See :ref:`api_paddle_nn_Bilinear` for details and output shape.
```

File: python/paddle/nn/layer/common.py [hunk 3]
```diff
@@ -1404,7 +1404,7 @@ class Embedding(Layer):
             default weight parameter property is used. See usage for details in :ref:`api_ParamAttr` . In addition,
             user-defined or pre-trained word vectors can be loaded with the :attr:`param_attr` parameter.
             The local word vector needs to be transformed into numpy format, and the shape of local word
-            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_initializer_NumpyArrayInitializer`
+            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_fluid_initializer_NumpyArrayInitializer`
```
Comment (ooooo-create) [inline python/paddle/nn/layer/common.py]:
```suggestion
            vector should be consistent with :attr:`num_embeddings` . Then :ref:`api_paddle_nn_initializer_Assign`
```

File: python/paddle/nn/functional/conv.py [hunk 4]
```diff
@@ -1067,7 +1067,7 @@ def conv2d_transpose(
     If bias attribution and activation type are provided, bias is added to
     the output of the convolution, and the corresponding activation function
     is applied to the final result.
-    See more detail in :ref:`api_nn_conv_ConvTranspose2d` .
+    See more detail in :ref:`api_paddle_nn_functional_conv2d_transpose` .
```
Comment (ooooo-create) [inline python/paddle/nn/functional/conv.py]:
```suggestion
    See more detail in :ref:`api_paddle_nn_Conv2DTranspose` .
```

File: python/paddle/nn/functional/conv.py [hunk 5]
```diff
@@ -1555,7 +1555,7 @@ def conv3d_transpose(
     If bias attribution and activation type are provided, bias is added to
     the output of the convolution, and the corresponding activation function
     is applied to the final result.
-    See more detail in :ref:`api_nn_conv_ConvTranspose3d` .
+    See more detail in :ref:`api_paddle_nn_functional_conv3d_transpose` .
```
Comment (ooooo-create) [inline python/paddle/nn/functional/conv.py]:
```suggestion
    See more detail in :ref:`api_paddle_nn_Conv3DTranspose` .
```

File: python/paddle/distributed/transpiler/distribute_transpiler.py [hunk 6]
```diff
@@ -157,8 +157,8 @@ class DistributeTranspilerConfig:
     .. py:attribute:: split_method (PSDispatcher)
 
           Methods of dispatching parameters for server,
-          :ref:`api_base_transpiler_RoundRobin` or
-          :ref:`api_base_transpiler_HashName` can be used and default is RoundRobin.
+          :ref:`api_fluid_transpiler_RoundRobin` or
```
Comment (ooooo-create) [inline python/paddle/distributed/transpiler/distribute_transpiler.py]: @sunzhongkai588 孙师傅来看看，我觉得本身这个文件的 api 就不是公开的，其次他引用的 api_label真实位置如下
image
，怎么处理呢，帮忙看看
- HashName
- BoundRobin
这两个都是

**Review by ooooo-create — COMMENTED** [review]
在 [Paddle 1.8 与 Paddle 2.0 API 映射表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/paddle_api_mapping_cn.html) 可以看到api迁移啦
屏幕截图 2023-09-22 185136
`api_paddle_nn_functional_conv2d_transpose` 这你写成了 api 本身的api_label
`api_paddle_nn_functional_conv3d_transpose` 同

**Comment by zade23** [comment]
@ooooo-create 感谢！
原来是API迁移了，以后做docathon会留一下迁移文档

File: python/paddle/distributed/transpiler/distribute_transpiler.py [hunk 7]
```diff
@@ -157,8 +157,8 @@ class DistributeTranspilerConfig:
...
+          :ref:`api_fluid_transpiler_HashName` can be used and default is RoundRobin.
```
Comment (sunzhongkai588) [inline python/paddle/distributed/transpiler/distribute_transpiler.py]: > 怎么处理呢，帮忙看看

那就不引用吧～ @ooooo-create @zade23 
如
```suggestion
          `RoundRobin` or
          `HashName` (both from `paddle.incubate.distributed.fleet.parameter_server.ir.ps_dispatcher`) can be used and default is RoundRobin.
```

File: python/paddle/nn/functional/activation.py [hunk 8]
```diff
@@ -136,7 +136,7 @@ def elu(x, alpha=1.0, name=None):
 def elu_(x, alpha=1.0, name=None):
     r"""
     Inplace version of ``elu`` API, the output Tensor will be inplaced with input ``x``.
-    Please refer to :ref:`api_nn_cn_elu`.
+    Please refer to :ref:`api_paddle_nn_ELU`.
```
Comment (sunzhongkai588) [inline python/paddle/nn/functional/activation.py]: 用 functional 下的 `elu` 比较合适。
```suggestion
    Please refer to :ref:`api_paddle_nn_functional_elu`.
```

File: python/paddle/nn/functional/activation.py [hunk 9]
```diff
@@ -774,7 +774,7 @@ def relu(x, name=None):
 def relu_(x, name=None):
     """
     Inplace version of ``relu`` API, the output Tensor will be inplaced with input ``x``.
-    Please refer to :ref:`api_nn_cn_relu`.
+    Please refer to :ref:`api_paddle_nn_ReLU`.
```
Comment (sunzhongkai588) [inline python/paddle/nn/functional/activation.py]: 同理，functional 下的 inplace api，refer 的应该也是functional下的原API
```suggestion
    Please refer to :ref:`api_paddle_nn_functional_relu`.
```

File: python/paddle/nn/functional/activation.py [hunk 10]
```diff
@@ -1248,7 +1248,7 @@ def softmax(x, axis=-1, dtype=None, name=None):
 def softmax_(x, axis=-1, dtype=None, name=None):
     r"""
     Inplace version of ``softmax`` API, the output Tensor will be inplaced with input ``x``.
-    Please refer to :ref:`api_nn_cn_softmax`.
+    Please refer to :ref:`api_paddle_nn_Softmax`.
```
Comment (sunzhongkai588) [inline python/paddle/nn/functional/activation.py]: 同理
```suggestion
    Please refer to :ref:`api_paddle_nn_functional_softmax`.
```

**Review by sunzhongkai588 — COMMENTED** [review]
除下述问题外，还发现了如下所示的问题（在[paddle.nn.Embedding](http://preview-paddle-pr-57630.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/Embedding_en.html)），即以 `p` 作为开头的部分 api label，仍存在不规范的问题。
image

这个 PR 先不修复这个问题吧。后续需要等该系列任务做完后，全量扫一下  :ref:`xxxx` 中 xxx 的内容，然后判断 xxx 是否和现有路径对的上。 @ooooo-create @SigureMo

File: python/paddle/nn/functional/activation.py (hunk 8 above)
Comment (zade23) [reply python/paddle/nn/functional/activation.py]: Down

**Review by sunzhongkai588 — APPROVED** [review]
LGTM
